### PR TITLE
Migrate package to Hypermodern python structure and guidelines

### PR DESCRIPTION
**Pull request by oncleben31**
Resources: https://cjolowicz.github.io/posts/hypermodern-python-01-setup/ & https://github.com/cjolowicz/cookiecutter-hypermodern-python

- cookiecutter template applied
- Sphinx documentation updated.
- Remove Python 3.5 and 3.6 support
- Add Python 3.9 support.
- Temporarily deactivate type hint check => will need rework in future PR

### CODE REVIEW

**Comment by Quentame** [comment]
Should solve #66 ? 🤔

**Comment by oncleben31** [comment]
> Should solve #66 ? 🤔

I think so but I'm not an expert. We will need to test to validate.

**Comment by Quentame** [comment]
Seems 2020.10.15 template

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+comment: false
+coverage:
+  status:
+    project:
+      default:
+        target: "100"
+    patch:
+      default:
+        target: "100"
```
Comment (Quentame) [inline codecov.yml:9]: Not sure if 100% coverage it possible here, maybe we should target 80%

**Comment by oncleben31** [comment]
Close this one will be done in multiple smaller PR.

**Pull request closed without merging**